src/views/docs-view/server.ts

```diff
@@ -283,8 +283,29 @@ export function getStaticGenerationFunctions<
 			if (currentRootDocsPath.visuallyHideSidebarTitle) {
 				docsSidebarLevel.visuallyHideTitle = true
 			}
-			// Add "Overview" item, unless explicitly disabled
-			if (currentRootDocsPath.addOverviewItem !== false) {
+
+			/**
+			 * Check the top level of the navData for "overview" items,
+			 * which are expected to be present for consistency.
+			 * If we do no have an overview item match, then we'll
+			 * automatically add an overview item.
+			 */
+			const overviewItemMatch = navDataWithFullPaths.find((item: MenuItem) => {
+				const isPathMatch =
+					item.path == '' ||
+					item.path == '/' ||
+					item.path == '/index' ||
+					item.path == 'index'
+				return isPathMatch
+			})
+			/**
+			 * Exception: If the first navData node is a `heading`,
+			 * we'll avoid adding an overview item even if there's
+			 * no overview item match.
+			 */
+			const firstItemIsHeading =
+				typeof navDataWithFullPaths[0]?.heading == 'string'
+			if (!overviewItemMatch && !firstItemIsHeading) {
```

Comment:
The `firstItemIsHeading` case applies to docs routes that are missing an `"Overview"` item, but would look weird if we added an `"Overview"` item since the first authored item in the navData is a `heading`. The [/terraform/internals](https://dev-portal-git-zsauto-overview-item-detect-hashicorp.vercel.app/terraform/internals) is one example (may be the only one, I'm not sure).

Related: I'm thinking it could make sense to use `firstItemIsHeading` to deprecate the `"visuallyHideSidebarTitle"` option on `rootDocsPath` in a similar way to what this PR does for `"addOverviewItem"`. Intent is to handle this in a separate PR.